File: backend/models/message.go
package models

import (
	"time"
	"gorm.io/gorm"
)

// Conversation 会话模型
type Conversation struct {
	ID        uint           `json:"id" gorm:"primarykey"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"-" gorm:"index"`
	
	// 参与者
	User1ID uint `json:"user1_id" gorm:"not null;index"`
	User2ID uint `json:"user2_id" gorm:"not null;index"`
	
	// 最后一条消息信息
	LastMessageID      *uint     `json:"last_message_id"`
	LastMessageContent string    `json:"last_message_content"`
	LastMessageTime    time.Time `json:"last_message_time"`
	
	// 未读消息数
	User1UnreadCount int `json:"user1_unread_count" gorm:"default:0"`
	User2UnreadCount int `json:"user2_unread_count" gorm:"default:0"`
	
	// 关联
	User1    User      `json:"user1" gorm:"foreignKey:User1ID"`
	User2    User      `json:"user2" gorm:"foreignKey:User2ID"`
	Messages []Message `json:"messages" gorm:"foreignKey:ConversationID"`
}

// Message 消息模型
type Message struct {
	ID        uint           `json:"id" gorm:"primarykey"`
	CreatedAt time.Time      `json:"created_at"`
	UpdatedAt time.Time      `json:"updated_at"`
	DeletedAt gorm.DeletedAt `json:"-" gorm:"index"`
	
	// 基本信息
	ConversationID uint   `json:"conversation_id" gorm:"not null;index"`
	SenderID       uint   `json:"sender_id" gorm:"not null;index"`
	ReceiverID     uint   `json:"receiver_id" gorm:"not null;index"`
	Content        string `json:"content" gorm:"type:text;not null"`
	MessageType    string `json:"message_type" gorm:"default:'text'"` // text, image, file
	
	// 状态
	IsRead bool `json:"is_read" gorm:"default:false"`
	
	// 关联
	Conversation Conversation `json:"conversation" gorm:"foreignKey:ConversationID"`
	Sender       User         `json:"sender" gorm:"foreignKey:SenderID"`
	Receiver     User         `json:"receiver" gorm:"foreignKey:ReceiverID"`
}

// TableName 指定表名
func (Conversation) TableName() string {
	return "conversations"
}

func (Message) TableName() string {
	return "messages"
}

File: clean_chat_app/lib/domain/entities/message.dart
import 'package:equatable/equatable.dart';
import 'user.dart';

/// 消息类型枚举
enum MessageType {
  text,
  image,
  file,
  system,
}

/// 消息实体
class Message extends Equatable {
  final int id;
  final int conversationId;
  final int senderId;
  final int receiverId;
  final String content;
  final MessageType messageType;
  final bool isRead;
  final DateTime createdAt;
  final DateTime updatedAt;
  final User? sender;
  final User? receiver;

  const Message({
    required this.id,
    required this.conversationId,
    required this.senderId,
    required this.receiverId,
    required this.content,
    this.messageType = MessageType.text,
    this.isRead = false,
    required this.createdAt,
    required this.updatedAt,
    this.sender,
    this.receiver,
  });

  /// 是否为文本消息
  bool get isTextMessage => messageType == MessageType.text;

  /// 是否为图片消息
  bool get isImageMessage => messageType == MessageType.image;

  /// 是否为文件消息
  bool get isFileMessage => messageType == MessageType.file;

  /// 是否为系统消息
  bool get isSystemMessage => messageType == MessageType.system;

  /// 复制并更新字段
  Message copyWith({
    int? id,
    int? conversationId,
    int? senderId,
    int? receiverId,
    String? content,
    MessageType? messageType,
    bool? isRead,
    DateTime? createdAt,
    DateTime? updatedAt,
    User? sender,
    User? receiver,
  }) {
    return Message(
      id: id ?? this.id,
      conversationId: conversationId ?? this.conversationId,
      senderId: senderId ?? this.senderId,
      receiverId: receiverId ?? this.receiverId,
      content: content ?? this.content,
      messageType: messageType ?? this.messageType,
      isRead: isRead ?? this.isRead,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
      sender: sender ?? this.sender,
      receiver: receiver ?? this.receiver,
    );
  }

  @override
  List<Object?> get props => [
        id,
        conversationId,
        senderId,
        receiverId,
        content,
        messageType,
        isRead,
        createdAt,
        updatedAt,
        sender,
        receiver,
      ];

  @override
  String toString() {
    return 'Message(id: $id, senderId: $senderId, content: $content, messageType: $messageType)';
  }
}